code/script.js

```diff
@@ -1,10 +1,203 @@
+const redish = "linear-gradient(90deg, rgba(84,10,10,1) 0%, rgba(64,25,10,1) 30%, rgba(92,76,19,1) 53%, rgba(96,112,26,1) 72%, rgba(91,64,21,1) 88%)";
+const yellowish = "linear-gradient(90deg, rgba(51,48,11,1) 0%, rgba(75,64,14,1) 30%, rgba(90,92,19,1) 53%, rgba(131,158,55,1) 72%, rgba(176,190,34,1) 88%)";
+const purpleish = "linear-gradient(90deg, rgba(48,11,51,1) 0%, rgba(21,14,75,1) 30%, rgba(75,19,92,1) 53%, rgba(78,26,70,1) 72%, rgba(131,27,49,1) 88%)";
+const backgroundGradients = [blueish, redish, greenish, yellowish, purpleish];
+
 
-// Global variables, if you need any, declared here
 
 // Functions declared here
 
+input.value = '';
+
+const greeting = () => {
+  showMessage(`Hello stranger! What's your name?`, 'bot');
+  input.value = '';
+};
+
+setTimeout(greeting, 500);
+
+
+//Called everytime a button is clicked and checks which number question is next
+const buttonClicked = () => {
+  
+  numberOfQuestionsAsked++; //adds 1 to the counter
+
+  if (numberOfQuestionsAsked < 6) {
+    if (numberOfQuestionsAsked === 1) {
+      showMessage(`My name is ${input.value}`, "user");
+      //saves the user's name in the object
+      allKnowingObj["name"] = input.value;
+      setTimeout(() => {
+        showMessage(`Hello ${allKnowingObj["name"]}! So you're wondering what your mood is? Let me help!
+        I'm going to ask you a couple of questions to help deterine the right color for you!`, "bot")
+      }, 500);
+      setTimeout(questionAge, 2000) //Calls on second question
+    } else if (numberOfQuestionsAsked === 2) {
+      showMessage(`I am ${allKnowingObj["age-range"]} years old`, "user");
+      setTimeout(questionEnergy, 500); //Calls on third question
+    } else if (numberOfQuestionsAsked === 3) {
+      showMessage(`I would say I'm at ${allKnowingObj["slider-value"]} out of 10`, "user");
+      setTimeout(questionMood(), 2000); //Calls on fourth question
+    } else if (numberOfQuestionsAsked === 4) {
+      showMessage(allKnowingObj["mood-smiley"], "user");
+      setTimeout(questionWeather, 500); //Calls on fifth question
+    } else if (numberOfQuestionsAsked === 5) {
+      showMessage(allKnowingObj["weather-choice"], "user");
+      setTimeout(questionHappy, 500); //Calls last question
+    };
+  } else { //Triggered if there have been 6 questions asked
+    setTimeout(() => {
+      showMessage("Thank you for your input! Here is your mood colour!", "bot");
+      if (allKnowingObj["happy"].includes("yes") || allKnowingObj["happy"].includes("oui")) { //if user is happy with the service they get a random colour form the gradient array
+        setTimeout(() => {
+          changeBackground(backgroundGradients[randomIndex()]);
+        }, 1500);
+      } else {
+        setTimeout(() => {
+          changeBackground("black"); //if user is not happy they get the colour black
```

Comment:
Oh my guess was right haha - fun to get to see the idea behind the color choices!